.github/workflows/pipeline.yml

```
@@ -116,27 +131,170 @@ jobs:
-          cd SavannaSync/backend/syncsenta-ai-service
-          mvn clean compile
-          echo "Backend built successfully"
+            cd SavannaSync/backend/syncsenta-ai-service
+            echo "Building backend with Maven..."
+            mvn clean package -DskipTests
+            echo "Backend built successfully"
+          '
+          rm -f private-key
+
+      - name: Copy backend JAR to deployment location
+        run: |
```

actionlint (1.7.8)

shellcheck reported issue in this script: SC2086:info:2:48: Double quote to prevent globbing and word splitting

(shellcheck)